modules/packages/Curl.chpl

```diff
@@ -769,9 +773,9 @@ module Curl {
 
         curl_easy_perform(curl);
 
-        extern proc strstr(haystack:c_string, needle:c_string):c_string;
+        extern proc strstr(haystack:c_ptrConst(c_char), needle:c_ptrConst(c_char)):c_ptrConst(c_char);
         // Does this URL accept range requests?
-        if strstr(buf.mem:c_string, c"Accept-Ranges: bytes"):c_ptr(void) == nil:c_ptr(void) {
+        if strstr(buf.mem:c_ptrConst(c_char), "Accept-Ranges: bytes".c_str()):c_ptr(void) == nil:c_ptr(void) {
```

Comment:
why? can't we implicitly convert param strings?

Reply:
removed the use of `.c_str()` on string literals in almost everywhere except for calls to `__primitive()`